uptrain/operators/language/sql.py

```diff
@@ -0,0 +1,185 @@
+"""
+Implement oeprators over sql data.
+"""
+
+from __future__ import annotations
+
+import itertools
+import json
+import os
+import typing as t
+
+import sqlglot
+from sqlglot.errors import ParseError
+from pydantic import BaseModel
+import polars as pl
+
+from uptrain.utilities.sql_utils import extract_tables_and_columns, extract_tables_and_columns_from_create, \
+    PLACEHOLDER_TABLE, execute_and_compare_sql
+
+if t.TYPE_CHECKING:
+    from uptrain.framework.base import *
+from uptrain.operators.base import *
+
+__all__ = ["HasStar", "GetSchemaDefinition", "ParseSQL", "ValidateTables", "ExecuteSQL"]
+
+
+# TODO: define a Table object and dump that into the dataframe
+class Table(BaseModel):
+    create_table_stmt: str
+    table_name: str
+    columns: list
+
+
+# operators to read schema definition from Spider sql dataset
+@register_op
+class GetSchemaDefinition(TableOp):
+    col_in_schema: str = "schema"
+    col_out: str = "schema_def"
+    col_out_tables: str = "schema_tables"
+    col_out_db_path: str = "db_path"
+    resources_path: str = ""
+
+    def setup(self, settings: t.Optional[Settings] = None):
+        self.resources_path = settings.resources_folder
+        return self
+
+    def __read_schema_definition(self, schema_name) -> (str, str, str):
+        # List to store all CREATE TABLE statements
+        create_table_statements = []
+        tables_and_columns = {}
+        db_path = os.path.join(self.resources_path, f'spider/database/{schema_name}/{schema_name}.sqlite')
```

Comment:
We need to generalize this so that it can work for non-spider tables also. DB path can be taken as input?

Reply:
This check is a specific to spider dataset. I am only using to add schema info, db path to my input set here https://github.com/uptrain-ai/uptrain/blob/03a5f83b71d7412a1ad251c0c2de3a0d1cae2156/examples/refactor/sql_test.py#L13-L29

Reply:
We can make this a helper function instead of a Check